### PR TITLE
Drawer navigation

### LINKED ISSUE #1: Create a Drawer Navigation
Until we get a decent UI, let's have a drawer navigation that shows all 4 levels and an about page.

### PR DESCRIPTION
**Pull request by ysawires**
8f31a2ab40c7431e0c87258c3c72ca4c8df69f46 is now a simple app with navigation for lessons.
3ec0f723f9e90719d659d85fb6c298f1b8251964 is config files that resolved several build errors. These changes were necessary for me.

### CODE REVIEW

File: android/app/build.gradle [hunk 1]
```diff
@@ -78,7 +78,7 @@ import com.android.build.OutputFile
  */
 
 project.ext.react = [
-    enableHermes: false,  // clean and rebuild if changing
+    enableHermes: true,  // clean and rebuild if changing
```
Comment (mark-yacoub) [inline android/app/build.gradle:81]: do you wanna clean and rebuild every single time?
what was not working?

File: android/app/src/main/java/com/metremenqeemi/MainApplication.java [hunk 2]
```diff
@@ -20,6 +22,11 @@ public boolean getUseDeveloperSupport() {
           return BuildConfig.DEBUG;
         }
 
+        @Override
+        protected JSIModulePackage getJSIModulePackage() {
+        return new ReanimatedJSIModulePackage();
+        }
+
```
Comment (mark-yacoub) [inline android/app/src/main/java/com/metremenqeemi/MainApplication.java:29]: if you added this from a guide somehwere, please add a comment why you're adding it, like what are you solving there.
and do this for everything you manually add

File: App.js [hunk 3]
```diff
@@ -1,110 +1,69 @@
...
-        style={[
-          styles.sectionDescription,
-          {
-            color: isDarkMode ? Colors.light : Colors.dark,
-          },
-        ]}>
-        {children}
-      </Text>
+    <View style={{flex: 1, alignItems: 'center', justifyContent: 'center'}}>
+      <Button onPress={() => navigation.navigate('Level Three')} title="Go to Level Three" />
```
Comment (mark-yacoub) [inline App.js]: so you for the single quotations right for navigate, but the string for `title` is `"`, your formatter shoould catch it.

File: App.js [hunk 4]
```diff
@@ -1,110 +1,69 @@
...
-import {
-  SafeAreaView,
-  ScrollView,
-  StatusBar,
-  StyleSheet,
-  Text,
-  useColorScheme,
-  View,
-} from 'react-native';
+function HomeScreen({navigation}) {
```
Comment (mark-yacoub) [inline App.js]: use anonymous functions everywhere, so this should be `HomeScreen = ({navigation}) => {....};`

File: android/app/src/main/java/com/metremenqeemi/MainApplication.java (hunk 2 above)
Comment (ysawires) [reply android/app/src/main/java/com/metremenqeemi/MainApplication.java:29]: This is in response to this error:
Error: Requiring module "node_modules/react-native-reanimated/src/reanimated2/index.js", which threw an exception: TypeError: null is not an object (evaluating 'InnerNativeModule.installCoreFunctions')

Solution from this github post:
https://github.com/software-mansion/react-native-reanimated/issues/846#issuecomment-943267584

File: android/app/build.gradle (hunk 1 above)
Comment (ysawires) [reply android/app/build.gradle:81]: This is in response to this error:
Error: Requiring module "node_modules/react-native-reanimated/src/reanimated2/index.js", which threw an exception: TypeError: null is not an object (evaluating 'InnerNativeModule.installCoreFunctions')

Solution from this github post:
software-mansion/react-native-reanimated#846 (comment)

Both this addition and the one in MainApplication.java seem to address the same issue.

**Review by mark-yacoub — APPROVED** [review]
tested and works now on ios

**Pull request merged**